src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -294,10 +358,123 @@ public override async ValueTask<IReadOnlyList<BulkStateItem>> GetBulkStateAsync(
                 Message = invokeRequest,
             };
 
-            return await this.MakeGrpcCallHandleError(
-                 options => client.InvokeServiceAsync(request, options),
-                 headers,
-                 cancellationToken);
+            var callOptions = new CallOptions(headers: headers ?? new Metadata(), cancellationToken: cancellationToken);
+
+            // add token for dapr api token based authentication
+            var daprApiToken = Environment.GetEnvironmentVariable("DAPR_API_TOKEN");
+
+            if (daprApiToken != null)
+            {
+                callOptions.Headers.Add("dapr-api-token", daprApiToken);
+            }
+
+            return (request, callOptions);
+        }
+
+        private async Task<InvocationResponse<TResponse>> MakeInvokeRequestAsyncWithResponse<TRequest, TResponse>(
+            InvocationRequest<TRequest> request,
+            CancellationToken cancellationToken = default)
+        {
+            
+            
+            Any serializedData = null;
+            if (request.Body != null)
+            {
+                if (typeof(TResponse) == typeof(byte[]))
```

Comment:
This is not a correct assumption. Serialization should  live outside this method.

It's totally valid to use `byte[]` with JSON serialization. It will result in a base64 encoded JSON string. I'd suggest that the callers of this method do the serialization.

Can you please add a test for `byte[]` with JSON so we don't regress this?

Reply:
I don't think the UT will catch anything because the server is mocked.Changed to instead make the raw format explicit rather than deducing from data type